hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/safemode/HealthyPipelineSafeModeRule.java

```diff
@@ -56,6 +57,8 @@ public class HealthyPipelineSafeModeRule extends SafeModeExitRule<Pipeline> {
   private final PipelineManager pipelineManager;
   private final int minHealthyPipelines;
   private final SCMContext scmContext;
+  private Set<PipelineID> pipelineIDSet;
```

Comment:
`pipelineIDSet` should be cleared in `cleanup`:

https://github.com/apache/ozone/blob/a05bfb9ad53ca745e75362729993b3f6083d61f1/hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/safemode/HealthyPipelineSafeModeRule.java#L187-L189

Also, to be consistent with `processedPipelineIDs`, I suggest `pipelineIDSet` to be `final`.  Instead of assignment in `initializeRule`, it could replace contents (`clear()`, `addAll()`).

Reply:
Done.